### DISCUSSION TITLE
Can't get the logAttribute of related model in placeholder of getDescriptionForEvent()

### DISCUSSION THREAD

**Discussion by JoaoAntonio2** [spatie/laravel-activitylog#808] [Q&A]
I have a Order model that belongsTo a Client model, when logging the Order model i want to retrieve a attribute "name" of the Client model so i made:

~~~
. . .
protected static $logAttributes = ['client.name'];
. . . 
~~~

and in the <code> getDescriptionForEvent </code> method i have:
~~~
public function getDescriptionForEvent(string $eventName): string
{
    if ($eventName == 'created') {
        return ':causer.name registerd the order :subject.code to the client :properties.attributes.client.name';
    }
}
~~~

But it simply doesnt bind the client name in the description correctly, the other placeholders works fine, but when i try to return the client.name it doesnt work.

Here is a Model instance of the Activity returned on tinker:

~~~
>>> Activity::latest()->first()->properties;
=> Illuminate\Support\Collection {#4360
     all: [
       "attributes" => [
         "id" => 163,
         "code" => "133512a",
         . . . 
         "created_at" => "2020-11-04T00:39:18.000000Z",
         "updated_at" => "2020-11-04T00:39:18.000000Z",
         "client.name" => "John Doe",
       ],
     ],
   }
~~~

**Comment by Gummibeer** [comment]
The issue here is the `Arr::get()` as it will try to find `$activity->properties['attributes']['client']['name']` but it should check for `$activity->properties['attributes']['client.name']`.
https://github.com/spatie/laravel-activitylog/blob/08a128edbfc96493f45d785381ab0be4479cd763/src/ActivityLogger.php#L221

To solve this the array should be saved using `Arr::set()` to transform the dot.notation key to a multi-level array.
https://github.com/spatie/laravel-activitylog/blob/08a128edbfc96493f45d785381ab0be4479cd763/src/Traits/DetectsChanges.php#L127

But this would be a BC. 🤔
I would like to ask @freekmurze how he would rate this - as a bug-fix or a breaking-change.

As a temporary fix we could try something like this to prefer an exact key-match in favor of the dot.notation magic.
```php
return $attributeValue[$propertyName] ?? Arr::get($attributeValue, $propertyName, $match);
```

**Reply by JoaoAntonio2** [reply]
Maybe a fix could be done replacing the dot with underscore when putting the new attribute to the array of attributes,

So instead of:
~~~
 "client.name" => "John Doe",
~~~
it could be:
~~~
 "client_name" => "John Doe",
~~~

and to avoid possibly conflicts with attributes that already exists we could be able to specify the key name in the <code>$logAttributes = ['client.name']</code> variable, that way:

~~~
protected static $logAttributes = ['client_fullname' => 'client.name'];
~~~

so this way the key will be named as "client_fullname", but the value will still be the attribute name of the Client related model.

~~~
 "client_fullname" => "John Doe",
~~~
